src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -238,7 +236,8 @@ protected override ImmutableArray<PendingBranch> RemoveReturns()
                 currentMethodOrLambda.IsAsync &&
                 !currentMethodOrLambda.IsImplicitlyDeclared)
             {
-                var foundAwait = result.Any(pending => pending.Branch != null && pending.Branch.Kind == BoundKind.AwaitExpression);
+                var foundAwait = result.Any(pending => pending.Branch != null &&
+                                                       pending.Branch.Kind == BoundKind.AwaitExpression);
```

Comment:
>pending.Branch.Kind == BoundKind.AwaitExpression [](start = 55, length = 48)

If you're going to change it you might consider `pending.Branch?.Kind == BoundKind.AwaitExpression`